crypto/sm2/sm2_crypt.c

```diff
@@ -166,19 +195,34 @@ int SM2_encrypt(const EC_KEY *key,
         msg_mask[i] ^= msg[i];
 
     if (EVP_DigestInit(hash, digest) == 0)
+    {
+        SM2err(SM2_F_SM2_ENCRYPT, ERR_R_EVP_LIB);
         goto done;
+    }
 
     if (EVP_DigestUpdate(hash, x2y2, field_size) == 0)
+    {
+        SM2err(SM2_F_SM2_ENCRYPT, ERR_R_EVP_LIB);
         goto done;
+    }
 
     if (EVP_DigestUpdate(hash, msg, msg_len) == 0)
+    {
+        SM2err(SM2_F_SM2_ENCRYPT, ERR_R_EVP_LIB);
         goto done;
+    }
 
     if (EVP_DigestUpdate(hash, x2y2 + field_size, field_size) == 0)
+    {
+        SM2err(SM2_F_SM2_ENCRYPT, ERR_R_EVP_LIB);
         goto done;
+    }
 
     if (EVP_DigestFinal(hash, C3, NULL) == 0)
+    {
+        SM2err(SM2_F_SM2_ENCRYPT, ERR_R_EVP_LIB);
         goto done;
+    }
 
     ctext_struct.C1x = x1;
     ctext_struct.C1y = y1;
```

Comment:
github won't let me comment on the actual lines. There are some function calls below here in this function that can fail which we should check the error returns for.